PostHog/posthog
# Improve insight loading state

## Changes

See #7153 

![2021-11-30 13 31 47](https://user-images.githubusercontent.com/984817/144057498-69559966-2204-488b-ac0e-53ebc06852e9.gif)

## How did you test this code?

adding unit tests and manually testing

1) set network tab in dev tools to throttle so you can see the impact
2) load an insight
3) change the chart type (eg. from linear to pie)
4) observe that only without this fix the pie chart briefly shows "NaN"
